content/file/file_unix_test.go

```diff
@@ -476,3 +476,39 @@ func TestStore_Dir_OverwriteSymlink_RemovalFailed(t *testing.T) {
 		t.Fatal("error calling Chmod(), error =", err)
 	}
 }
+
+func Test_extractTarDirectory_PreservePermissions(t *testing.T) {
```

Comment:
Forgot to mention, since this function is defined in `utils.go`, it's better to test it in `utils_unix_test.go` instead of `file_unix_test.go`.
A side note: we need to include the oras copyright header at the beginning of each file.